sycl/plugins/esimd_emulator/pi_esimd_emulator.hpp

```diff
@@ -104,48 +110,96 @@ struct _pi_mem : _pi_object {
 
   pi_context Context;
 
+  // To be used for piEnqueueMemBufferMap
   char *MapHostPtr = nullptr;
 
-  // Mutex for load/store accessing
+  // Mutex for load/store accessing to CM-managed surfaces from memory
+  // instrinsic implementations
   std::mutex mutexLock;
 
-  // Surface index used by CM
-  int SurfaceIndex;
+  // Surface index
+
+  unsigned int SurfaceIndex;
+  // Supplementary data to keep track of the mappings of this memory
+  // created with piEnqueueMemBufferMap
+  struct Mapping {
+    // The offset in the buffer giving the start of the mapped region.
+    size_t Offset;
+    // The size of the mapped region.
+    size_t Size;
```

Comment:
Uninitialized values. May be ok or may be not ok, it depends.

Reply:
This is imported from level_zero. Will update if necessary.

Reply:
Done.